OpenRA.Mods.Common/Activities/Air/Land.cs

```diff
@@ -129,15 +129,23 @@ public override bool Tick(Actor self)
 				}
 			}
 
-			// Move towards landing location
-			if (aircraft.Info.VTOL && (pos - targetPosition).HorizontalLengthSquared != 0)
+			// Move towards landing location/facing
+			if (aircraft.Info.VTOL)
 			{
-				QueueChild(new Fly(self, Target.FromPos(targetPosition)));
+				if ((pos - targetPosition).HorizontalLengthSquared != 0)
+				{
+					QueueChild(new Fly(self, Target.FromPos(targetPosition)));
 
-				if (desiredFacing != -1)
-					QueueChild(new Turn(self, desiredFacing));
+					if (desiredFacing != -1)
+						QueueChild(new Turn(self, desiredFacing));
 
-				return false;
+					return false;
+				}
+				else if (desiredFacing != -1 && desiredFacing != aircraft.Facing)
+				{
+					QueueChild(new Turn(self, desiredFacing));
```

Comment:
Is there any harm in queuing the turn even if facing == desired? IIRC there isn't, and we can then reduce duplication by having the check/queue in the `if` case above outside of the loop.

Reply:
There is.

We can't queue the `Turn` only inside the `if ((pos - targetPosition).HorizontalLengthSquared != 0)`, because that's the very source of #16819 (on bleed `Turn` isn't queued if `HorizontalLengthSquared == 0`).
And if we leave it outside the above `if`, we can't queue it _un_conditionally, because then `Land` would go into an infinite loop.

Reply:
The alternative would be to leave this entire `if` block like on bleed, minus the conditional `Turn`, and then add this `else if` as an `if` in the last `(if Info.VTOL)` block below that so far only handles the descent.